compiler/noirc_evaluator/src/brillig/mod.rs

```diff
@@ -74,7 +77,7 @@ impl std::ops::Index<FunctionId> for Brillig {
 impl Ssa {
     /// Compile Brillig functions and ACIR functions reachable from them
     #[tracing::instrument(level = "trace", skip_all)]
-    pub(crate) fn to_brillig(&self, enable_debug_trace: bool) -> Brillig {
+    pub(crate) fn to_brillig(&mut self, enable_debug_trace: bool) -> Brillig {
```

Comment:
This makes me a little sad.

Reply:
If it's necessary it's necessary but I guess it raises the question on whether we need to store the global usage tracker on the `Ssa` struct.

Reply:
I could avoid it by cloning `self.used_globals` if we desire, that is really the only reason I did this. I could also do it outside of `to_brillig`.

Reply:
I made a `to_brillig` and a `to_brillig_with_globals` method that takes in the used globals map. Let me know if that is preferred.